### PR TITLE
[Model] Replace diffusers apply_rotary_emb with omni RotaryEmbedding

### PR DESCRIPTION
**Pull request by iwzbi**
## Purpose
resolve https://github.com/vllm-project/vllm-omni/issues/471

### CODE REVIEW

**Comment by hsliuustc0106** [comment]
any test results compared with torch native implmentation in terms of acc and speed

**Comment by hsliuustc0106** [comment]
is there any test result comparison for different kernels

**Comment by iwzbi** [comment]
> is there any test result comparison for different kernels

sorry, testing in progress

**Comment by iwzbi** [comment]
> is there any test result comparison for different kernels
- peformance benchmark:

```bash
python3 benchmarks/diffusion_benchmark_serving.py \
  --model /mnt/share_data/wz/Ovis-Image-7B/ \
  --dataset vbench --task t2i --num-prompts 10 \
  --port 8099 --num-inference-steps 28

```

before
image

with new kernel
image

- accuracy test:

```python
import torch

from vllm_omni.entrypoints.omni import Omni

if __name__ == "__main__":
    generator = torch.Generator("cuda").manual_seed(42)
    model_name = "/mnt/share_data/wz/Ovis-Image-7B/"
    m = Omni(model=model_name, output_type="latent", dtype=torch.half)

    # high resolution may cause OOM on L4
    height = 256
    width = 256
    images = m.generate(
        "a cat wearing sunglasses, cyberpunk style",
        height=height,
        width=width,
        dtype=torch.half,
        num_inference_steps=1,
        guidance_scale=0,
        generator=generator,
        num_outputs_per_prompt=1,
    )
    for i in images:
        print(i.images)

```
print output latent:

before
image

with new kernel
image

**Comment by hsliuustc0106** [comment]
the kernels show significant difference in acc? does it affect the e2e visual results?

**Comment by iwzbi** [comment]
> the kernels show significant difference in acc? does it affect the e2e visual results?

The pictures look the same. 1024 * 1024 ，inference 28 steps

before
image

with new kernel
image

**Comment by ZJY0516** [comment]
@tjtanaa The amd ci (docker image build) failed. Could you please take a look?

> ERROR: failed to solve: process "/bin/sh -c python3 -m pip uninstall -y vllm && rm -rf vllm &&    git clone https://github.com/vllm-project/vllm.git &&     cd vllm &&     git checkout ${VLLM_VERSION} &&     python3 -m pip install -r requirements/rocm.txt &&     python3 setup.py clean --all &&     PYTORCH_ROCM_ARCH=${PYTORCH_ROCM_ARCH} python3 setup.py develop &&     cd ../ &&     rm -rf vllm/.git" did not complete successfully: exit code: 1

I was wondering if we could install directly from a binary wheel instead of building from source.

**Comment by iwzbi** [comment]
> LGTM. Could you please help verify the functionality of the cache-dit feature?

sure. benchmark with cachedit using same parameters:
image

**Comment by ZJY0516** [comment]
> > LGTM. Could you please help verify the functionality of the cache-dit feature?
> 
> sure. benchmark with cachedit using same parameters: image

What about generation quality

**Comment by iwzbi** [comment]
> > > LGTM. Could you please help verify the functionality of the cache-dit feature?
> > 
> > 
> > sure. benchmark with cachedit using same parameters: image
> 
> What about generation quality

looks a bit blurry

image

**Comment by ZJY0516** [comment]
> > > > LGTM. Could you please help verify the functionality of the cache-dit feature?
> > > 
> > > 
> > > sure. benchmark with cachedit using same parameters: image
> > 
> > 
> > What about generation quality
> 
> looks a bit blurry
> 
> image

cache is lossy acceleration method. results should be the same with or without caching enabled.

**Comment by hsliuustc0106** [comment]
tthanks
